js/app.js

```diff
@@ -99,33 +100,95 @@ function renderProducts() {
   allProducts[productThree].views++;
 }
 
+// function to count the clicks generated for each imageOne, Two or Three of the sameNumbers array. Then it takes those 3 images out of the array and stops allowing clicks when the clicksAllowed is reached.
 function clickHandler(event) {
   clicks++;
   let imageClicked = event.target.title;
-  if (imageClicked === 'imageOne'){
+  if (imageClicked === 'imageOne') {
     allProducts[sameNumbers[0]].clicks++;
   }
-  if (imageClicked === 'imageTwo'){
+  if (imageClicked === 'imageTwo') {
     allProducts[sameNumbers[1]].clicks++;
   }
-  if (imageClicked === 'imageThree'){
+  if (imageClicked === 'imageThree') {
     allProducts[sameNumbers[2]].clicks++;
   }
-  sameNumbers = sameNumbers.slice(3,6);
+  sameNumbers = sameNumbers.slice(3, 6);
   if (clicks === clicksAllowed) {
     myContainer.removeEventListener('click', clickHandler);
     myButton.addEventListener('click', resultHandler);
   }
   renderProducts();
 }
 
+// function for pushing products as strings into labels for chart.js
+function productsForChart() {
+  let productNames = [];
+  for (let i = 0; i < allProducts.length; i++) {
+    productNames.push(allProducts[i].name);
+  }
+  return productNames;
+}
+
+// function to put views into chart
+function viewsForChart() {
+  let productViews = [];
+  for (let i = 0; i < allProducts.length; i++) {
+    productViews.push(allProducts[i].views);
+  }
+  return productViews;
+}
+
+// function to put clicks into chart
+function clicksForChart() {
+  let productClicks = [];
+  for (let i = 0; i < allProducts.length; i++) {
+    productClicks.push(allProducts[i].views);
```

Comment:
looks like if you change views to clicks in this clicksForChart function, you'll be set!  let me know if you have any questions!